### PR TITLE
Cleanup CQ and cohort resource stats

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Follow up to #2037.
The only information we need from the cohort to calculate a dominant resource share is how much lendable resources there are.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/cache/clusterqueue.go [hunk 1]
```diff
@@ -75,8 +75,8 @@ type ClusterQueue struct {
 	// deleted, or the resource groups are changed.
 	AllocatableResourceGeneration int64
 
-	// ResourceStats holds nominal quota and usage for the resources of the ClusterQueue, independent of the flavor.
-	ResourceStats ResourceStats
```
Comment (mimowo) [inline pkg/cache/clusterqueue.go:79]: It seems we can now drop type declarations for `ResourceStats` and `QuotaStats`

**Review by mimowo — COMMENTED** [review]
LGTM, one nit

File: pkg/cache/clusterqueue.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/cache/clusterqueue.go:79]: True. Done

**Pull request merged**